apps/designer/app/designer/features/style-panel/style-section.tsx

```diff
@@ -0,0 +1,777 @@
+          // }}
+        />
+        <Unit value={value} />
+      </Flex>
+    </Grid>
+  );
+};
+
+const SelectControl = ({
+  currentStyle,
+  inheritedStyle,
+  setProperty,
+  styleConfig,
+}: ControlProps) => {
+  // @todo show which instance we inherited the value from
+  const value = getFinalValue({
+    currentStyle,
+    inheritedStyle,
+    property: styleConfig.property,
+  });
+
+  if (value === undefined) return null;
+
+  const setValue = setProperty(styleConfig.property);
+
+  return (
+    <Flex align="center">
+      <PropertyName
+        property={styleConfig.property}
+        label={styleConfig.label}
+        css={{
+          fontWeight: "500",
+          marginRight: "$sizes$1",
+        }}
+      />
+      <Select
+        options={styleConfig.items.map(({ label }) => label)}
+        value={String(value.value)}
+        onChange={setValue}
+        ghost
+        css={{
+          gap: "calc($sizes$1 / 2)",
+          px: "$sizes$1",
+          fontWeight: "500",
+          textTransform: "capitalize",
+          "&:hover": { background: "none" },
+        }}
+      />
+    </Flex>
+  );
+};
+
+const IconButtonWithMenuControl = ({
+  currentStyle,
+  inheritedStyle,
+  setProperty,
+  styleConfig,
+  category,
+}: ControlProps) => {
+  const value = getFinalValue({
+    currentStyle,
+    inheritedStyle,
+    property: styleConfig.property,
+  });
+  const isCurrentBreakpoint = useIsFromCurrentBreakpoint(styleConfig.property);
+
+  if (value === undefined) return null;
+
+  const setValue = setProperty(styleConfig.property);
+  const currentValue = value.value as string;
+
+  if (String(currentStyle.display?.value).includes("flex") !== true) {
+    return (
+      <ComboboxControl
+        currentStyle={currentStyle}
+        inheritedStyle={inheritedStyle}
+        setProperty={setProperty}
+        styleConfig={styleConfig}
+        category={category}
+      />
+    );
+  }
+
+  return (
+    <IconButtonWithMenu
+      label={styleConfig.label}
+      items={styleConfig.items}
+      value={String(currentValue)}
+      onChange={setValue}
+      icons={
+        (
+          icons as unknown as Record<
+            string,
+            Record<string, (props: unknown) => JSX.Element>
+          >
+        )[styleConfig.property]
+      }
+      css={{
+        ...(styleConfig.property !== "flexDirection" && {
+          svg: {
+            transform: `rotate(${
+              currentStyle.flexDirection?.value === "column"
+                ? 90 * (styleConfig.property === "alignItems" ? -1 : 1)
+                : 0
+            }deg)`,
+          },
+        }),
+        ...(isCurrentBreakpoint && {
+          color: "$colors$blue11",
+          backgroundColor: "$colors$blue4",
+          "&:not([data-state=open]):hover": {
+            backgroundColor: "$colors$blue4",
+          },
+        }),
+      }}
+    ></IconButtonWithMenu>
+  );
+};
+
+const GridControl = ({
```

Comment:
same thing, gridControl should be just the glue code between state and rendering